EcdsaTestCase::test_roundtrip fails with ecdsa-0.15

Hi,

It looks like EcdsaTestCase::test_roundtrip fails on newer versions of ecdsa; jsonpickle 1.4.1 w/ ecdsa 0.14.1 passes, while jsonpickle 1.4.1 w/ ecdsa-0.15 fails.

Full build log can be found [here](https://github.com/jsonpickle/jsonpickle/files/4905132/build.txt), but here is a snippet:
```
============================================================================= FAILURES =============================================================================
___________________________________________________________________ EcdsaTestCase.test_roundtrip ___________________________________________________________________

self = <ecdsa_test.EcdsaTestCase testMethod=test_roundtrip>

    def test_roundtrip(self):
        if self.should_skip:
            return self.skip('ecdsa module is not installed')
    
        message = 'test'.encode('utf-8')
        key_pair = self.SigningKey.generate(curve=self.NIST384p)
        sig = key_pair.sign(message)
    
        serialized = jsonpickle.dumps(key_pair.get_verifying_key())
        restored = jsonpickle.loads(serialized)
>       self.assertTrue(restored.verify(sig, message))

tests/ecdsa_test.py:32: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3.7/site-packages/ecdsa/keys.py:602: in verify
    return self.verify_digest(signature, digest, sigdecode, True)
/usr/lib/python3.7/site-packages/ecdsa/keys.py:648: in verify_digest
    r, s = sigdecode(signature, self.pubkey.order)
/usr/lib/python3.7/site-packages/ecdsa/util.py:312: in sigdecode_string
    l = orderlen(order)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

order = {'py/object': 'builtins.mpz'}

    def orderlen(order):
>       return (1+len("%x" % order))//2  # bytes
E       TypeError: %x format: an integer is required, not dict

/usr/lib/python3.7/site-packages/ecdsa/util.py:40: TypeError
========================================================================= warnings summary =========================================================================
tests/numpy_test.py::NumpyTestCase::test_dtype_roundtrip
  /var/tmp/portage/dev-python/jsonpickle-1.4.1-r1/work/jsonpickle-1.4.1/tests/numpy_test.py:56: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
    np.dtype('1i4', align=True),

tests/numpy_test.py::NumpyTestCase::test_ndarray_roundtrip
  /var/tmp/portage/dev-python/jsonpickle-1.4.1-r1/work/jsonpickle-1.4.1/jsonpickle/ext/numpy.py:292: UserWarning: ndarray is defined by reference to an object we do not know how to serialize. A deep copy is serialized instead, breaking memory aliasing.
    warnings.warn(msg)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
===================================================================== short test summary info ======================================================================
FAILED tests/ecdsa_test.py::EcdsaTestCase::test_roundtrip - TypeError: %x format: an integer is required, not dict
====================================================== 1 failed, 267 passed, 21 skipped, 2 warnings in 7.61s =======================================================
 * ERROR: dev-python/jsonpickle-1.4.1-r1::gentoo failed (test phase):
 *   Tests failed with python3.7
 * 
```

I've also attached some system information [here](https://github.com/jsonpickle/jsonpickle/files/4905128/system-info.txt). Please let me know if I can provide any more information.

Thanks!

--- a/docs/extensions.rst
+++ b/docs/extensions.rst
@@ -10,3 +10,12 @@ enable the numpy extension by registering its handlers::
 
     >>> import jsonpickle.ext.numpy as jsonpickle_numpy
     >>> jsonpickle_numpy.register_handlers()
+
+Ecdsa
+-----
+For the ecdsa module's keys, when trying to serialize them with
+``gmpy2`` installed, jsonpickle will error unless the ``gmpy``
+handlers are registered:
+
+    >>> import jsonpickle.ext.gmpy as jsonpickle_gmpy
+    >>> jsonpickle_gmpy.register_handlers()

--- a/jsonpickle/ext/gmpy.py
+++ b/jsonpickle/ext/gmpy.py
@@ -1,0 +1,22 @@
+import gmpy2 as gmpy
+
+from ..handlers import BaseHandler, register, unregister
+
+__all__ = ['register_handlers', 'unregister_handlers']
+
+
+class GmpyMPZHandler(BaseHandler):
+    def flatten(self, obj, data):
+        data['int'] = int(obj)
+        return data
+
+    def restore(self, data):
+        return gmpy.mpz(data['int'])
+
+
+def register_handlers():
+    register(gmpy.mpz, GmpyMPZHandler, base=True)
+
+
+def unregister_handlers():
+    unregister(gmpy.mpz)

--- a/jsonpickle/pickler.py
+++ b/jsonpickle/pickler.py
@@ -48,7 +48,7 @@ def encode(
         instances. If you experience (de)serialization being incorrect when you
         use numpy, pandas, or sklearn handlers, this should be set to ``False``.
         If you want the output to not include the dtype for numpy arrays, add
-        ``jsonpickle.register(numpyp.generic,
+        ``jsonpickle.register(numpy.generic,
          UnpicklableNumpyGenericHandler, base=True)`` before your pickling code.
     :param max_depth: If set to a non-negative integer then jsonpickle will
         not recurse deeper than 'max_depth' steps into the object.  Anything

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -2,6 +2,7 @@ bson
 coverage<5
 ecdsa
 feedparser
+gmpy2
 numpy
 pandas
 pymongo

--- a/setup.cfg
+++ b/setup.cfg
@@ -59,6 +59,7 @@ testing =
 	# local
 	ecdsa
 	feedparser
+	gmpy2
 	numpy
 	pandas
 	pymongo

--- a/tests/ecdsa_test.py
+++ b/tests/ecdsa_test.py
@@ -3,6 +3,7 @@ from __future__ import absolute_import, division, unicode_literals
 
 import unittest
 
+import pytest
 from helper import SkippableTest
 
 import jsonpickle

--- a/tests/ecdsa_test.py
+++ b/tests/ecdsa_test.py
@@ -8,6 +8,14 @@ from helper import SkippableTest
 import jsonpickle
 
 
+@pytest.fixture(scope='module', autouse=True)
+def gmpy_extension():
+    """Initialize the gmpy extension for this test module"""
+    jsonpickle.ext.gmpy.register_handlers()
+    yield  # control to the test function.
+    jsonpickle.ext.gmpy.unregister_handlers()
+
+
 class EcdsaTestCase(SkippableTest):
     def setUp(self):
         try:

